compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -3681,19 +3681,36 @@ class MatchReducer(initctx: Context) extends TypeComparer(initctx) {
 
             stableScrut.member(typeMemberName) match
               case denot: SingleDenotation if denot.exists =>
-                val info = denot.info match
-                  case alias: AliasingBounds           => alias.alias        // Extract the alias
-                  case ClassInfo(prefix, cls, _, _, _) => prefix.select(cls) // Re-select the class from the prefix
-                  case info => info // Notably, RealTypeBounds, which will eventually give a MatchResult.NoInstances
-                val info1 = stableScrut match
+                val info = stableScrut match
                   case skolem: SkolemType =>
-                    dropSkolem(info, skolem).orElse:
-                      info match
-                        case info: TypeBounds  => info                       // Will already trigger a MatchResult.NoInstances
-                        case _                 => RealTypeBounds(info, info) // Explicitly trigger a MatchResult.NoInstances
-                  case _ => info
-                rec(capture, info1, variance = 0, scrutIsWidenedAbstract)
+                    /* If it is a skolem type, we cannot have class selections nor
+                     * abstract type selections. If it is an alias, we try to remove
+                     * any reference to the skolem from the right-hand-side. If that
+                     * succeeds, we take the result, otherwise we fail as not-specific.
+                     */
+                    def adaptToTriggerNotSpecific(info: Type): Type = info match
```

Comment:
It looks like this comment refers to the code below starting with `denot.info match`, not to the behavior of the def itself.

Reply:
It's about the whole body of the `case SkolemType`. I added a blank line, which hopefully makes it not misleading anymore?

Reply:
Ah I see, usually we write those comments with `//`.

Reply:
I thought `/*` was just the multi-line version of `//`, whereas `/**` is the doc-of-the-`def` thing.

Reply:
Yeah using `/*` is fine, just uncommon in this codebase I think

Reply:
I always doc'd local defs with `/*` seeing as the docs don't appear anywhere in the scaladocs anyways, to not be misleading as to that.  It's funny how that thought process leads to unintended confusion in this case 😄